lgc/builder/BuilderImpl.cpp

```diff
@@ -334,6 +335,67 @@ BranchInst *BuilderImpl::createIf(Value *condition, bool wantElse, const Twine &
   return branch;
 }
 
+// A simple memory efficient container that holds up to 64 instructions in a bit vector. It needs two helper data
+// structures: 1. indexInsnMap that maps an instruction to its index in the bit vector and 2. insnChain that maps an
+// index back to an instruction.
+class TinyInsnSet {
+  BitVector bits;
+  const DenseMap<Instruction *, unsigned> &indexInsnMap;
+  const SmallVector<Instruction *, 64> &insnChain;
+
+public:
+  TinyInsnSet(const DenseMap<Instruction *, unsigned> &indexInsnMap, const SmallVector<Instruction *, 64> &insnChain)
+      : indexInsnMap(indexInsnMap), insnChain(insnChain) {
+    bits.resize(64);
+  }
+
+  class iterator {
+    BitVector::const_set_bits_iterator it;
+    const SmallVector<Instruction *, 64> &insnChain;
+
+  public:
+    iterator(BitVector::const_set_bits_iterator it, const SmallVector<Instruction *, 64> &insnChain)
+        : it(it), insnChain(insnChain) {}
+    iterator &operator++() {
+      ++it;
+      return *this;
+    }
+    Instruction *operator*() {
```

Comment:
Can you please separate the member functions with empty lines?

Reply:
Done